common/src/main/java/com/microsoft/identity/common/internal/request/AdalBrokerRequestAdapter.java

```diff
@@ -0,0 +1,319 @@
+                AuthenticationConstants.Broker.ACCOUNT_CORRELATIONID
+        );
+        if (TextUtils.isEmpty(correlationIdString)) {
+            UUID correlationId = UUID.randomUUID();
+            correlationIdString = correlationId.toString();
+        }
+        parameters.setCorrelationId(correlationIdString);
+
+        String resource = bundle.getString(
+                AuthenticationConstants.Broker.ACCOUNT_RESOURCE
+        );
+        Set<String> scopes = new HashSet<>();
+        scopes.add(getScopeFromResource(resource));
+        parameters.setScopes(scopes);
+
+        final String clientId = bundle.getString(
+                AuthenticationConstants.Broker.ACCOUNT_CLIENTID_KEY
+        );
+        parameters.setClientId(clientId);
+
+        parameters.setForceRefresh(Boolean.parseBoolean(
+                bundle.getString(AuthenticationConstants.Broker.BROKER_FORCE_REFRESH))
+        );
+
+        parameters.setClaimsRequest(
+                bundle.getString(AuthenticationConstants.Broker.ACCOUNT_CLAIMS)
+        );
+
+        parameters.setLoginHint(
+                bundle.getString(AuthenticationConstants.Broker.ACCOUNT_NAME)
+        );
+
+        final List<Pair<String, String>> extraQP = getExtraQueryParamAsList(
+                bundle.getString(AuthenticationConstants.Broker.ACCOUNT_EXTRA_QUERY_PARAM)
+        );
+        parameters.setExtraQueryStringParameters(extraQP);
+
+        return parameters;
+    }
+
+    private String getPackageNameFromBundle(final Bundle bundle, final Context context) {
+        String packageName = bundle.getString(AuthenticationConstants.Broker.CALLER_INFO_PACKAGE);
+        if (TextUtils.isEmpty(packageName)) {
+            packageName = bundle.getString(AuthenticationConstants.AAD.APP_PACKAGE_NAME);
+            if (TextUtils.isEmpty(packageName)) {
+                packageName = context.getPackageName();
+            }
+        }
+        return packageName;
+    }
+
+    /**
+     * Helper to get Extra QP as a List (V2 format) from String (adal format)
+     */
+    private List<Pair<String, String>> getExtraQueryParamAsList(@Nullable final String extraQueryParamString) {
+        final List<Pair<String, String>> extraQPList = new ArrayList<>();
+        if (!StringUtil.isEmpty(extraQueryParamString)) {
+            final String[] extraQueryParams = extraQueryParamString.split("&");
+
+            for (final String param : extraQueryParams) {
+                if (!StringUtil.isEmpty(param)) {
+                    String[] split = param.split("=");
+                    final String name = split[0];
+                    final String value = (split.length > 1) ? split[1] : null;
+                    final Pair<String, String> extraQPPair = new Pair<>(name, value);
+                    extraQPList.add(extraQPPair);
+                }
+            }
+        }
+        return extraQPList;
+    }
+
+
+    /**
+     * TODO : Refactor to remove this code and move the logic to better place
```

Comment:
Can this be removed now or is this a follow-up item?

Reply:
follow up item , issue here : https://github.com/AzureAD/microsoft-authentication-library-common-for-android/issues/436